packages/@aws-cdk/aws-cloud9-alpha/test/integ.owner.ts

```diff
@@ -0,0 +1,34 @@
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { Stack, App, StackProps, CfnOutput } from 'aws-cdk-lib';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import { Construct } from 'constructs';
+import * as cloud9 from '../lib';
+
+class Cloud9Env extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const vpc = new ec2.Vpc(this, 'VPC', {
+      restrictDefaultSecurityGroup: false,
+      maxAzs: 2,
+      natGateways: 1,
+    });
+
+    // create a cloud9 ec2 environment in a new VPC
+    const c9env = new cloud9.Ec2Environment(this, 'C9Env', {
+      vpc,
+      imageId: cloud9.ImageId.AMAZON_LINUX_2,
+      owner: cloud9.Owner.assumedRole('807336301584', 'Admin/test'),
```

Comment:
Hey, just to confirm, this is not your AWS account id or a valid one?

Reply:
Seems to be somewhat of a convention to use `123456789012` for tests?

Reply:
It is an actual `accountId` and an actual user in that account. If I understood integration tests in the CDK correctly, they synth, deploy and destroy and actual stack in an actual environment to verify correctness.

In my case, I need to refer to existing IAM entities in the account that I run the integration tests in and I don't see any other way.

I could avoid having to specify the account id by using `Stack.of(this).account`, but that would require a breaking change to `environment.ts:Owner` as the functions are static and not members of the `Ec2Environment` construct. This would also make assumed-roles that are defined in other accounts impossible (although I don't think it's this is possible)

Please advise if you see another way to test this

Reply:
But these integ tests would fail if the resource is deleted from the account or the permissions no longer allow operations on the account :)
Also, it is not recommended usually to hardcode account ids in code base. 

----

What you can do here is,
1. Create a role in the integ test and that should be a dependency for provisioning your C9 Environment.
2. Use the [environment](https://docs.aws.amazon.com/cdk/v2/guide/environments.html) from the CLI which has the account and the region. For instance,
```
new MyDevStack(app, 'dev', { 
  env: { 
    account: process.env.CDK_DEFAULT_ACCOUNT, 
    region: process.env.CDK_DEFAULT_REGION 
}});
````

Let me know if this makes sense.

Reply:
Also, would `accountId` ever be different from the user's account id?